app/controllers/photoTagController.js

```diff
@@ -1,7 +1,42 @@
+import { PhotoTagCreate, PhotoTagResponse } from "../models/apiModels/photoTag.js";
+import PhotoTagAccessor from "../databaseAccessors/photoTagAccessor.js";
+import { ErrorRepeatedTagName, ErrorUnexpected, HttpError } from "../error/errors.js";
+
+
 /**
  * PhotoTagController Class
  *
  * This class controls the behaviour of any web request
  * related to PhotoTags
  */
-export default class PhotoTagController {}
+export default class PhotoTagController {
+  /**
+   * Creates a PhotoTag with ref fields populated
+   *
+   * @param {Request} req
+   * @param {Response} res
+   */
+  static async create(req, res) {
+    try {
+      const body = new PhotoTagCreate(req.body);
+      
+      // if creating a tag that already exists
+      const listOfTags = await PhotoTagAccessor.getAllTags();
+      if (listOfTags.map((tag) => tag.tagName).includes(body.tagName)) {
+        throw new ErrorRepeatedTagName();
+      } 
```

Comment:
Since we use Mongoose schema, we have set `tagName` to be unique, meaning that if you do try to insert a duplicate record, it will automatically throw, no need to get every single tagName, and iterate through each and every one of them to see if the new one exists. Think of the case when we have 500+ tags. This will take a lot of time, and isn't efficient whenever we need to create a new tag. Its computationally taxing.

Instead, you can just try to insert, and catch a mongodb error.